chunky/src/java/se/llbit/chunky/renderer/FrameSampler.java

```diff
@@ -0,0 +1,173 @@
+package se.llbit.chunky.renderer;
+
+import org.apache.commons.math3.util.FastMath;
+import se.llbit.chunky.renderer.scene.Scene;
+import se.llbit.chunky.resources.BitmapImage;
+import se.llbit.math.ColorUtil;
+import se.llbit.math.QuickMath;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.concurrent.BlockingQueue;
+
+/**
+ * A FrameSampler decides how samples are taken and reconstructs the image from them.
+ */
+public abstract class FrameSampler {
+  public static final byte SAMPLER_TYPE_UNIFORM = 1;
```

Comment:
This should be an enum instead.

Reply:
Good call, initially I just needed the numeric values for loading and saving render dumps but the enum will be useful in the `Scene` too, to specify the sampling strategy.